lazy_theta_star_planner/include/lazy_theta_star_planner/lazy_theta_star_planner.h

```diff
@@ -0,0 +1,75 @@
+// Copyright 2020 Anshumaan Singh
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef LAZY_THETA_STAR_PLANNER__LAZY_THETA_STAR_PLANNER_H_
+#define LAZY_THETA_STAR_PLANNER__LAZY_THETA_STAR_PLANNER_H_
+
+#include <iostream>
+#include <cmath>
+#include <string>
+#include <chrono>
+#include <queue>
+#include <algorithm>
+#include <memory>
+#include <vector>
+#include "rclcpp/rclcpp.hpp"
+#include "nav2_core/global_planner.hpp"
+#include "nav_msgs/msg/path.hpp"
+#include "nav2_util/robot_utils.hpp"
+#include "nav2_util/lifecycle_node.hpp"
+#include "nav2_costmap_2d/costmap_2d_ros.hpp"
+#include "nav2_costmap_2d/cost_values.hpp"
+#include "nav2_util/node_utils.hpp"
+#include "lazy_theta_star_planner/lazy_theta_star2.h"
+
+namespace lazyThetaStarPlanner
+{
+
+class LazyThetaStarPlanner : public nav2_core::GlobalPlanner
+{
+public:
+  std::shared_ptr<tf2_ros::Buffer> tf_;
+  nav2_util::LifecycleNode::SharedPtr node_;
+  nav2_costmap_2d::Costmap2D * costmap_;
+
+  std::string globalFrame_, name_;
+  int how_many_corners_;
+  double interpolation_dist_;
+  int lethal_cost_;
+  std::unique_ptr<lazyThetaStar::LazyThetaStar> planner_;
+
+  unsigned int src_[2], dst_[2];
```

Comment:
Does this need to be stored at all or can this just be local variables in the `makePlan` function?

Reply:
That's additional state information I don't think is good to have laying around

Reply:
Since I am directly using the costmap sent to the plannerServer, for the algorithm, by pointing it by a pointer, so I need the pointer to the algorithm's class, will make it private though

Reply:
But for the start / goal why do those need to be class variables at all. The `planner_->plan()` function could be given the start / goal or you could make a `setStart()` and `setGoal()` methods to send them over as private variables

Reply:
I realised that, and made those functions in the version I am working on, and also made the variables private